src/components/navbar/navbar.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from "react";
+import { NavLink } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBarsStaggered } from "@fortawesome/free-solid-svg-icons";
+
+const Navbar = () => {
+  const [toggleMenu, setToggleMenu] = useState(false);
+  const menuRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleClickOutSide = (event: MouseEvent) => {
+      if (menuRef.current && !menuRef.current.contains(event.target as Node)) {
+        setToggleMenu(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutSide);
+
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutSide);
+    };
+  }, []);
+
+  console.log(toggleMenu);
+
+  const Links = [
+    {
+      to: "/",
+      label: "Stellar",
+    },
+    {
+      to: "/planatrip",
+      label: "Plan a trip",
+    },
+    {
+      to: "/discover",
+      label: "Discover",
+    },
+    {
+      to: "/checkflights",
+      label: "Check flights",
+    },
+  ];
+
+  const LinksTwo = [
+    {
+      to: "/login",
+      label: "Login",
+    },
+    {
+      to: "/register",
+      label: "Register",
+    },
+  ];
+
+  const Menu = [
+    {
+      to: "/planatrip",
+      label: "Plan a trip",
+    },
+    {
+      to: "/discover",
+      label: "Discover",
+    },
+    {
+      to: "/checkflights",
+      label: "Check flights",
+    },
+    {
+      to: "/login",
+      label: "Login",
+    },
+    {
+      to: "/register",
+      label: "Register",
+    },
+  ];
+
+  const linkClasses: string =
+    "invisible md:visible text-white lg:text-green cursor-pointer text-xs leading-7 font-robo font-bold hover:text-red-700 transition-colors duration-200 delay-75 mr-5 lg:mr-10";
+  const logoClasses: string =
+    "text-white cursor-pointer text-2xl -tracking-wider leading-7 font-noto font-bold mr-10 lg:mr-20 ";
+  const loginClasses: string =
+    "invisible md:visible text-white cursor-pointer text-xs leading-7 font-robo font-bold hover:text-red-700 transition-colors duration-200 delay-75";
+  const menuClasses: string =
+    "flex flex-col text-xs items-center my-4 text-white leading-7 font-robo font-bold ";
+
```

Comment:
Great work moving these to constants, much easier for maintenance well done. 👏 👏